### ISSUE TITLE
bug: ion-button aria-description does not reactively change

### ISSUE THREAD

**Issue by anilanar** [ionic-team/ionic-framework#30626]
### Prerequisites

- [x] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [x] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [x] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v8.x

### Current Behavior

With React, if we have a dynamic aria-description such as `<IonButton aria-description={count} />`, the `button` inside the shadow dom gets stuck with the initial `count`. 

### Expected Behavior

`aria-description` and pretty much all attributes that are "copied" from the host to the native button inside the shadow DOM should be reactive.

### Steps to Reproduce

1. Go to https://stackblitz.com/edit/dx7kxrku?file=src%2Fmain.tsx
2. Open Devtools
3. Observe that the native button inside `ion-button` has `aria-description="0"` even though the component increments it every second

### Code Reproduction URL

https://stackblitz.com/edit/dx7kxrku?file=src%2Fmain.tsx

### Ionic Info

I don't know what `ionic info` is. I don't have such a command available anywhere, neither in our project nor on Stackblitz.

**Comment by ReneZeidler** [comment]
This also applies to other elements which can act as buttons, like `<ion-item>`, `<ion-card>`, etc.
